src/modules/services/reducers.js

```diff
@@ -236,11 +241,46 @@ export const serviceDataTypes = (
             };
         }
 
+        // Handle itemsByDatasetID
+        case LOADING_SERVICE_DATA_TYPES_BY_DATASET.BEGIN:
+            return {...state, isFetchingItemsByDatasetID: true};
+
+        case LOADING_SERVICE_DATA_TYPES_BY_DATASET.END:
+        case LOADING_SERVICE_DATA_TYPES_BY_DATASET.TERMINATE:
+            return {...state, isFetchingItemsByDatasetID: false};
+
+        case FETCH_SERVICE_DATA_TYPES_BY_DATASET.RECEIVE: {
+            const { datasetID, data } = action;
+            const existingItems = state.itemsByDatasetID[datasetID] || [];
+            return {
+                ...state,
+                itemsByDatasetID: {
+                    ...state.itemsByDatasetID,
+                    [datasetID]: [...existingItems, ...data],
```

Comment:
Comment added!

Reply:
as we discussed before, this action is not idempotent - meaning it doesn't give the same result if it is called again. a way to fix this would be to further break this down by either data type ID (nested object) and then optionally also create a flattened version but from this object so there arent duplicate values.